certbot/util.py

```diff
@@ -278,76 +277,52 @@ def get_filtered_names(all_names):
     return filtered_names
 
 
-def get_os_info(filepath="/etc/os-release"):
+def get_os_info():
     """
     Get OS name and version
 
-    :param str filepath: File path of os-release file
     :returns: (os_name, os_version)
     :rtype: `tuple` of `str`
     """
 
-    if os.path.isfile(filepath):
-        # Systemd os-release parsing might be viable
-        os_name, os_version = get_systemd_os_info(filepath=filepath)
-        if os_name:
-            return os_name, os_version
-
-    # Fallback to platform module
-    return get_python_os_info()
+    return (distro.id(), distro.version(pretty=False))
 
 
-def get_os_info_ua(filepath="/etc/os-release"):
+def get_os_info_ua():
     """
     Get OS name and version string for User Agent
 
-    :param str filepath: File path of os-release file
     :returns: os_ua
     :rtype: `str`
     """
 
-    if os.path.isfile(filepath):
-        os_ua = get_var_from_file("PRETTY_NAME", filepath=filepath)
-        if not os_ua:
-            os_ua = get_var_from_file("NAME", filepath=filepath)
-        if os_ua:
-            return os_ua
+    return distro.version(pretty=True)
```

Comment:
`distro.version` seems pretty different from our current behavior to me. The value in our user agent will change from "'Ubuntu 18.04.3 LTS" to "18.04 (bionic)" for example.

Instead, what about calling `distro.name(pretty=True)` and if that is an empty string call `" ".join(get_python_os_info(full_name=True))`? See my comment about adding that function back if you haven't already.

Reply:
Distro does fallback on different version information sources if `/etc/os-release` isn't available. How strict is the requirement to keep version information intact for UA strings? For example, `distro.name(pretty=True)` on RHEL6 returns: `Red Hat Enterprise Linux Server release 6.10 (Santiago)`, which is not exactly the same (even though close) to the one we compile from `platform.linux_distribution()` output.

Reply:
There's no real requirement for them to not change. I think the only things to think about would be:

1. Are we collecting significantly more information?
2. Changing the values may make it a little harder for us to group users based on their UA.

I personally don't think either of these are changing enough with the current plan to make it a problem.

Reply:
I don't think we're collecting more information. The format just might change a bit. The main source for our UA info has been `/etc/os-release` PRETTY_NAME, or if that's not found, NAME. We have this from distro again, but as distro compiles multiple fingerprinting sources we might not have as much control over it.